src/CMakeLists.txt

```diff
@@ -579,6 +579,7 @@ endif()
 if (TARGET ch_rust::skim)
     # Add only -I, library is needed only for clickhouse-client/clickhouse-local
     dbms_target_include_directories(PRIVATE $<TARGET_PROPERTY:ch_rust::skim,INTERFACE_INCLUDE_DIRECTORIES>)
+    dbms_target_link_libraries(PUBLIC ch_rust::skim)
```

Comment:
@rschu1ze Robert, could you please take a look? I'm trying to repro this issue with GNU ld, but no luck. See original issue in the PR description.
@bkuschel what ld version do you use?

Reply:
```bash
> ld --version
GNU ld (GNU Binutils for Ubuntu) 2.38

```bash
> uname -a
Linux xxx #66-Ubuntu SMP Fri Jan 20 14:30:43 UTC 2023 s390x s390x s390x GNU/Linux
```

You need to install rust and cargo. I haven't tried it on a x86 since `llvm` is used by default and i've never tried overriding it. I don't imagine it being specific to s390x but you could try building it in a `s390x/ubuntu:22.04` image (using docker multiarch support) and you should see it.

Reply:
thanks Boris,  I was able to repro this

Reply:
Same :-) `cmake -B build -S . -DENABLE_RUST=1 -DLINKER_NAME=/usr/bin/ld`

Can also confirm that the fix resolves the linking issue. If all CI builds are okay, then green light from my side.

Reply:
...it's very strange though, why clang can link it but gnu ld can't...